flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Table.java

```diff
@@ -1355,6 +1355,33 @@ default Table limit(int offset, int fetch) {
      */
     TableResult executeInsert(String tablePath, boolean overwrite);
 
+    /**
+     * Declares that the pipeline defined by the given {@link Table} object should be written to a
+     * table defined by a {@link TableDescriptor}. It executes the insert operation.
+     *
+     * <p>The {@link TableDescriptor descriptor} is registered as an inline (i.e. anonymous)
+     * temporary table (see {@link TableEnvironment#createTemporaryTable(String, TableDescriptor)})
+     * using a unique identifier. Note that calling this method multiple times, even with the same
+     * descriptor, results in multiple sinks being registered.
+     *
+     * <p>Examples:
+     *
+     * <pre>{@code
+     * Schema schema = Schema.newBuilder()
+     *   .column("f0", DataTypes.STRING())
+     *   .build();
+     *
+     * Table table = tableEnv.from(TableDescriptor.forConnector("datagen")
+     *   .schema(schema)
+     *   .build());
+     *
+     * table.executeInsert(TableDescriptor.forConnector("blackhole")
+     *   .schema(schema)
+     *   .build());
+     * }</pre>
+     */
+    TableResult executeInsert(TableDescriptor descriptor);
```

Comment:
can you also add a `executeInsert(String tablePath, boolean overwrite)` descriptor method